private/buf/buffetch/buffetch.go

```diff
@@ -360,3 +360,178 @@ func NewWriter(
 type getSourceBucketOptions struct {
 	workspacesDisabled bool
 }
+
+type RefBuilder interface {
+	// GetGitRef returns a Ref for a Git repository.
+	GetGitRef(
+		ctx context.Context,
+		format string,
```

Comment:
Why do these need formats?

Reply:
The `format` is passed to the internal constructor for the specific Refs, such as `internal.NewGitRef` and `internal.NewModuleRef`. 

They are used for error messages, for example: [`NewInvalidPathError(format, path)`](https://github.com/bufbuild/buf/blob/main/private/buf/buffetch/internal/module_ref.go#L41).
So that in V2, I pass in `module` and the error message will look like `invalid path xyz for module`.
And for parsing command line input argument, `mod` is passed and the error mesage will look like `invalid path xyz for mod`.